File: 01_basics/01_05_palindrome/palindrome.cpp
#include<iostream>

using namespace std;

bool is_palindrome(int x)
{
    if (x < 0)
    {
        return false;
    }
    long long reversed = 0;
    long long temp = x;

    while (temp != 0)
    {
        int digit = temp % 10;
        reversed = reversed * 10 + digit;
        temp /= 10;
    }

    return (reversed == x);
}

int main()
{
    int num = 123321;

    cout << "Is " << num << " Palindrome: " << is_palindrome(num) << endl;
}

File: 01_basics/01_05_palindrome/Palindrome.java
public class Palindrome {
    public static void main(String[] args) {
        int num = 123321;

        System.out.println(String.format("Is %d Palindrome: %b", num, isPalindrome(num)));
    }

    private static boolean isPalindrome(int x) {
        String numberInString = String.valueOf(x);
        Integer stringLength = numberInString.length();

        for(Integer index = 0; index < ((stringLength)/2); index++) {
            if(numberInString.charAt(index) != numberInString.charAt(stringLength - index - 1)) {
                return false;
            }
        }

        return true;
    }
}
